panacea-3/v2.0.5/upgrade.md

```diff
@@ -0,0 +1,130 @@
+    |Tendermint|[v0.34.14](https://github.com/tendermint/tendermint/releases/tag/v0.34.14)|[v0.34.21](https://github.com/tendermint/tendermint/releases/tag/v0.34.21)|
+
+
+## Changes
+
+- Upgrades Cosmos SDK from v0.42.11 to v0.45.9 which contains many new features, bug fixes, and especially a [IBC security fix](https://forum.cosmos.network/t/ibc-security-advisory-dragonberry/7702).
+- Removes the `x/token` module which is not used at all.
+
+## Guides for Validators and Full-Node Operators
+
+If this SoftwareUpgrade proposal is approved, the state machine of the `panacead` will be stopped as soon as the chain reaches to the block height 7700000.
+It means that new blocks will not be produced until the `panacead` daemon is restarted with the new version.
+
+When the state machine is stopped, you will be able to see the following error logs:
+```
+ERR UPGRADE "v2.0.5" NEEDED at height: 7700000:
+ERR CONSENSUS FAILURE!!! err="UPGRADE \"v2.0.5\" NEEDED at height: 7700000: "
+```
+
+Then, stop your process.
+```bash
+pkill panacead
+
+# or by other commands depending on your environment
+```
+
+After that, you should build a new binary as below. And, please replace the old `panacead` binary with the new one.
+
+```bash
+git clone https://github.com/medibloc/panacea-core
+cd panacea-core
+git checkout v2.0.5
+make install
+
+panacead version
+# should print v2.0.5
+```
+
+(Recommended) Before starting the new `panacead` process, please back up the `~/.panacea` directory just in case.
+```bash
+cp -R ~/.panacea ~/panacea-3-v2.0.3-backup
+
+# or, to the external storage
+```
+
+Then, you can start the `panacead` again.
+```bash
+panacead start
+
+# or by other commands depending on your environment
+```
+
+NOTE:
+If you are using the [Cosmovisor](https://medibloc.gitbook.io/panacea-core/guide/cosmovisor) process manager, please build the new `panacead` binary manually and put that under the `$HOME/.panacea/cosmovisor/upgrades/v2.0.5/bin/` by following the [guide](https://medibloc.gitbook.io/panacea-core/guide/cosmovisor#cosmovisor-setup). The [auto-download](https://github.com/cosmos/cosmos-sdk/tree/main/cosmovisor#auto-download) is not supported yet because the appropriate version of the [libwasmvm.so](https://github.com/CosmWasm/wasmvm/blob/v0.14.0/api/libwasmvm.so) must be installed as well. Instead of installing the `libwasmvm.so` separately, I would recommend you build the `panacea-core` by the guide above. Then, the `libwasmvm.so` will be installed automatically.
+
+NOTE: Since Cosmos SDK v0.43, the new configuration [grpc-web](https://github.com/cosmos/cosmos-sdk/blob/2582f0aab7b2cbf66ade066fe570a4622cf0b098/server/config/toml.go#L182) has been added to the `config.toml`. This configuration is enabled by default with a port 9091. If you don't want this feature, please disable this or change the port.
```

Comment:
It has been added to `app.toml`.
```suggestion
NOTE: Since Cosmos SDK v0.43, the new configuration [grpc-web](https://github.com/cosmos/cosmos-sdk/blob/2582f0aab7b2cbf66ade066fe570a4622cf0b098/server/config/toml.go#L182) has been added to the `app.toml`. This configuration is enabled by default with a port 9091. If you don't want this feature, please disable this or change the port.
```